src/C-interface/dense/bml_parallel_dense_typed.c

```diff
@@ -26,7 +26,7 @@ void TYPED_FUNC(
     bml_allGatherVParallel_dense) (
     bml_matrix_dense_t * A)
 {
-#ifdef DO_MPI
+#ifdef BML_MPI_NONDIST
```

Comment:
You are decoupling `DO_MPI` from `BML_MPI_NONDIST` and should at the very least do something like
```
#if defined(DO_MPI) && defined(BML_MPI_NONDIST)
```
to ensure that the code is compiled with MPI support.